src/main/java/com/blog/web/controllers/ArticleController.java

```diff
@@ -0,0 +1,122 @@
+    }
+
+    @GetMapping("/articles/{articleId}")
+    public String showArticle(@PathVariable("articleId") long articleId, Model model) {
+        ArticleDto articleDto = articleService.findArticleById(articleId);
+        model.addAttribute("article", articleDto);
+        UserEntity user = userService.getLoggedInUser();
+        model.addAttribute("user", user);
+        return "articles/show";
+    }
+
+    @GetMapping("/articles/new")
+    public String createArticleForm(Model model) {
+        UserEntity user = userService.getLoggedInUser();
+        model.addAttribute("user", user);
+        final Article article = new Article();
+        model.addAttribute("article", article);
+        return "articles/new";
+    }
+
+    @PostMapping("/articles/new")
+    public String saveArticle(@Valid @ModelAttribute("article") ArticleDto articleDto,
+                              BindingResult result,
+                              Model model)
+    {
+        // if un-logged in user tries to create an article
+        // redirect them to login page
+        UserEntity user = userService.getLoggedInUser();
+        if(userService.getLoggedInUser().getId() == null) {
+            return "redirect:/userlogin";
+        }
+        else if(result.hasErrors()) {
+            model.addAttribute("article", articleDto);
+            return "articles/new";
+        }
+        else {
+            articleService.saveArticle(articleDto);
+            return "redirect:/articles";
+        }
+    }
+
+
+    @GetMapping("/articles/delete/{articleId}")
+    public String deleteArticle(@PathVariable("articleId") Long articleId) {
+        UserEntity user = userService.getLoggedInUser();
+        ArticleDto article = articleService.findArticleById(articleId);
+        UserEntity owner = article.getCreatedBy();
+        if(owner.equals(user)) {
+            articleService.delete(articleId);
+        }
+        return "redirect:/articles";
+    }
+
+    @GetMapping("/articles/edit/{articleId}")
+    public String editArticleForm(@PathVariable("articleId") long articleId, Model model) {
+        UserEntity user = userService.getLoggedInUser();
+        model.addAttribute("user", user);
+        ArticleDto articleDto = articleService.findArticleById(articleId);
+        model.addAttribute("article", articleDto);
+        return "articles/edit";
+    }
+
+    @PostMapping("/articles/edit/{articleId}")
+    public String updateArticle(@PathVariable("articleId") Long articleId,
+                                @Valid @ModelAttribute("article") ArticleDto article,
+                                BindingResult result) {
+        if(result.hasErrors()) {
+            return "articles/edit";
+        }
+        article.setId(articleId);
+        articleService.updateArticle(article);
+        return "redirect:/articles";
+    }
+
+    @GetMapping("/articles/search")
+    public String searchArticle(@RequestParam(value = "search") String search, Model model) {
```

Comment:
this might be cleaner for this function:
https://www.baeldung.com/spring-pathvariable